src/_articles/archive/zones.md

```diff
@@ -165,16 +171,24 @@ Another way to produce an uncaught error is
 to call `new Future.error()` or
 a Completer's `completeError()` method.
 
-Use the `onError` argument to `runZoned()` to
-install a _zoned error handler_—an asynchronous error handler
+`runZoned()` does not handle errors.
+Its error-handling parameter `onError` 
+was deprecated because it requires the
+function's return type to always be nullable,
+which is unneccesarily cumbersome for [Null Safe code][]. 
+
+[Null Safe code]: /null-safety/index
+
+Instead, use `runZonedGuarded()` to install a
+_zoned error handler_ — an asynchronous error handler
```

Comment:
Might be worth saying that it catches both uncaught async errors *and* uncaught synchronous errors (which is why it needs the nullability in the return type, something throwing synchronously doesn't have value to return).

In comparison `runZoned` *can* have an error handler (because you can install one as part of the  `ZoneSpecification`), but that only works for asynchronous errors.

Let's take a step back.

The title of this section is "handling async errors".

The text here is mainly about *uncaught* async errors and zones handling those, which makes sense since the file is about zones.
Maybe the title should include "uncaught", otherwise it seems like simple error handling like `try`/`catch` in an `async` function would also be relevant. (And even if only looking at zones, `Zone.errorCallback` might be relevant too, and I don't think we need to burden readers with that. It's too specialized.)

Focusing on uncaught errors, and handling them, the section can then either show *all* the ways to do that, or it can focus on the most common one.
Which one do we want?

"All the ways" include `runZoned`, `runZonedGuarded`, and `Zone.fork`, and mentions `ZoneSpecification.uncaughtErrorHandler`.

I think I'd just focus on `runZonedGuarded`, as the easiest and most convenient way, and refer only to the others in a "see also" section.

In which case, we don't need to compare `runZoned` and `runZonedGuarded`, just say how `runZonedGuarded` works. (Even if we do mention the others, having the section above starting out telling you what *not* to do seems like wasted prime real-estate, it's better to start saying what you _should_ do. Then the reader can stop at any point after that and still come away with the answer they need.)

So, maybe:

> One of the most used features of zones is
> their ability to catch and handle uncaught errors.
>
>  An _uncaught error_ is often caused by some code using `throw`
> to raise an exception that is not handled by a `catch` statement.
> In an `async` function, such an error usually ends up as the result of a `Future`,
> which allows you to handle the error where you `await` the future.
> Sometimes an error escapes, and sometimes a future is not `await`ed, 
> in which case it becomes an uncaught error.
>
> Such an error can crash your program.
>
> A zone can have an _uncaught error handler_ installed when the zone is created.
> If an uncaught error occurs in code which runs in that zone, 
> the uncaught error handler is called instead of just crashing the program.
> (In fact, the code which crashes the program on an uncaught error
> is just the uncaught error handler of the root zone, 
> which any zone inherits if it doesn’t specify another handler).
>
> The easiest way to introduce a new zone with an uncaught error handler
> is to run code using `runZoneGuarded`. It’s `onError` callback becomes the
> uncaught error handler of a new zone, and it’s *also* used to handle any
> synchronous errors thrown by the call.
>
> Example:
>
> ... the timer example ...

In any case, don't focus on what doesn't work, just say what does, and how to do it.
Then maybe add a section saying

> See also `Zone.fork`, `Zone.runGuarded` and `ZoneSpecification.uncaughtErrorHandler`.

Reply:
@lrhn I only slightly reworded some parts of this. It's not necessary because I don't think I changed any meaning, but if you have a minute to confirm [this section](https://github.com/dart-lang/site-www/pull/4403/commits/5c1173dfdc70060289eb1b280be6991f3b48cfed#diff-2b7e547663296c3749c4b011b531d9163e186954912d493b1ecaa60063b013abR157) is still accurate that would be great thank you!